projects/distributed-tracing/src/shared/dashboard/dashboard-wrapper/navigable-dashboard.component.ts

```diff
@@ -67,6 +72,14 @@ export class NavigableDashboardComponent implements OnChanges {
         defaultIfEmpty(this.defaultJson)
       );
     }
+
+    if (changeObject.filterConfig && this.hasFilterBarConfig()) {
+      this.attributes$ = this.metadataService.getFilterAttributes(this.filterConfig!.filterBar!.scope);
```

Comment:
Pretend `!` doesn't exist, pretty much always. If the compiler can't figure it out, it generally means we could do a better job typing (with a few exceptions like map and array access). Here, one issue would be that if I were to set filterConfig to undefined, we'd leave `attributes$` set to something incorrect. Instead, we could write this as something like
```
    if (changeObject.filterConfig) {
      this.attributes$ = this.filterConfig?.filterBar ? this.metadataService.getFilterAttributes(this.filterConfig.filterBar.scope) : of([]);
    }
```